### ISSUE TITLE
Validating intuition of training progress before investing in full fine-tuning effort

### ISSUE THREAD

**Issue by leebean337** [MarvinLvn/voice-type-classifier#14]
We've fine-tuned the VTC model for a modest 6 epoches to ensure its working as expected before investing in our full fine-tuning effort. We noticed a few progress metrics that we were not sure we understood fully and wanted to check with you that they were not a source of concern before we proceeded. Specifically, we executed the following training command (using your updated config.yml file):
```
pyannote-audio mlt train --subset=train --to=106 --from=100 model_ellis6_sad/ ELLIS.SpeakerDiarization.Classroom
```
and during training we noted that the denominator reported in the progress bar (10,800) and the duration of training (~7 hours) for the initial epoch (see screenshot below) was considerably more than for the subsequent epoches (101 through 106), where the denominator was 1,350 and duration of training was less than 1 hour each.  
![image](https://user-images.githubusercontent.com/46361085/87748886-05925580-c7ac-11ea-8bfe-028e5b3a3c80.png)

Would you be able to confirm that this is the expected behavior and possibly provide some intuition why this would be the case? Our best guess is that it may have to do with GPU caching of the calculations but we're not certain.

Separately, but potentially related, when we apply our new (validated) model on the test set, we observed the following progress output:
![image](https://user-images.githubusercontent.com/46361085/87749195-d4665500-c7ac-11ea-879b-c200a095020f.png)
This was a little confusing given that it appears 6 separates models were used for prediction (which aligns to our 6 sets of model weights resulting from our training) however our understanding was that the validate step would select the single best performing model and apply that alone. So we are unclear why the 'apply' command would assess all 6 models, and assuming this is the case, we see once again a dramatic difference in duration of time required for this assessment between the first interation and subsequent ones (9min:45sec for the first versus less than 2 minutes for the rest).

We are hoping to launch our extended training (50+ epoches) over the weekend but want to ensure these initial trials do not indicate a lingering configuration problem.

**Comment by MarvinLvn** [comment]
Quickly answering now, as I won't be available for next 2 weeks.

```
During training we noted that the denominator reported in the progress bar (10,800) and the duration of training (~7 hours) for the initial epoch (see screenshot below) was considerably more than for the subsequent epoches (101 through 106), where the denominator was 1,350 and duration of training was less than 1 hour each.
```

The progress bar you sent is the scheduler of the learning rate. The model is gonna be ran for a few (fake) epochs at the beginning of the training (and every 14 epochs) to choose an "optimal" learning rate. This is decorrelated from the epoch duration that you can observe after.

```
This was a little confusing given that it appears 6 separates models were used for prediction (which aligns to our 6 sets of model weights resulting from our training) however our understanding was that the validate step would select the single best performing model and apply that alone. So we are unclear why the 'apply' command would assess all 6 models, and assuming this is the case, we see once again a dramatic difference in duration of time required for this assessment between the first interation and subsequent ones (9min:45sec for the first versus less than 2 minutes for the rest).
```

For inference time : 
First iteration : the model is applied on the validation set.
Then the thresholds (computed during validation) for each of the classes are applied (those are the next 5 iterations).

From what I can see, everything seems normal. You can have a look at the files generated during the validation/inference time to see if anything seems weird (the onset/offset parameters for instance).

**Comment by leebean337** [comment]
Thank you for the clarification!
